### PR TITLE
feat(rust): add missing CRUD options for `ockam trust-context` command

### PR DESCRIPTION
**Pull request by adrianbenavides**

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/trust_context/delete.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+}
+
+fn run_impl(opts: CommandGlobalOpts, cmd: DeleteCommand) -> crate::Result<()> {
+    let DeleteCommand { name } = cmd;
+    let state = opts.state.trust_contexts;
+    match state.get(&name) {
+        // If it exists, proceed
+        Ok(_) => {
+            if let ConfirmResult::No = opts.terminal.confirm(&fmt_warn!(
```
Comment (etorreborre) [inline implementations/rust/ockam/ockam_command/src/trust_context/delete.rs:41]: Should we add a `--yes` option every time we require a user confirmation?

**Review by etorreborre — DISMISSED** [review]
🚗

File: implementations/rust/ockam/ockam_command/src/trust_context/delete.rs (hunk 1 above)
Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_command/src/trust_context/delete.rs:41]: Yeah... I think we should have the same as we currently have in the `ockam reset` command:
- TTY: ask for confirmation if `--yes` is not passed
- Non TTY: fail if `--yes` is not passed

Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_command/src/trust_context/delete.rs:41]: Yeah... I think we should have the same as we currently have in the `ockam reset` command:
- TTY: ask for confirmation if `--yes` is not passed
- Non TTY: fail if `--yes` is not passed

I'll tackle this in a separate PR

Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_command/src/trust_context/delete.rs:41]: I've created an issue for that https://github.com/build-trust/ockam/issues/5041

File: implementations/rust/ockam/ockam_command/src/trust_context/show.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        }
+    }
+}
+
+fn run_impl(opts: CommandGlobalOpts, cmd: ShowCommand) -> crate::Result<()> {
+    let name = cmd
+        .name
+        .unwrap_or(opts.state.trust_contexts.default()?.name().to_string());
+    let state = opts.state.trust_contexts.get(&name)?;
+    println!("Trust context:");
```
Comment (caoakleyii) [inline implementations/rust/ockam/ockam_command/src/trust_context/show.rs]: Let's continue to use `opts.terminal.write_line` for these? 

(I have a seperate pr that update it to not write in cases of tty, and json output)

File: implementations/rust/ockam/ockam_command/src/trust_context/show.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    for line in state.to_string().lines() {
+        println!("{:2}{}", "", line)
```
Comment (caoakleyii) [inline implementations/rust/ockam/ockam_command/src/trust_context/show.rs]: `opts.terminal`

File: implementations/rust/ockam/ockam_command/src/trust_context/list.rs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            eprintln!("{e}");
+            std::process::exit(e.code());
+        }
+    }
+}
+
+fn run_impl(opts: CommandGlobalOpts) -> crate::Result<()> {
+    let states = opts.state.trust_contexts.list()?;
+    if states.is_empty() {
+        return Err(anyhow!("No trust contexts registered on this system!").into());
```
Comment (caoakleyii) [inline implementations/rust/ockam/ockam_command/src/trust_context/list.rs]: The error changes are merged into develop now, if you want to go ahead and update that in this PR.

**Review by caoakleyii — DISMISSED** [review]
Few small non-blocking changes.

File: implementations/rust/ockam/ockam_command/src/trust_context/show.rs (hunk 2 above)
Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_command/src/trust_context/show.rs]: Sure, I thought I did it, my bad!

**Comment by adrianbenavides** [comment]
@CAOakleyII I've replaced `anyhow` by `miette` and refactored the `show` and `list` output handling.

**Review by etorreborre — APPROVED** [review]

**Pull request merged**